src/calendar/internal.tsx

```diff
@@ -39,8 +39,12 @@ export default function Calendar({
   const gridWrapperRef = useRef<HTMLDivElement>(null);
   const [focusedDate, setFocusedDate] = useState<Date | null>(null);
 
+  const valueDateCache = useDateCache();
+  const focusedDateCache = useDateCache();
```

Comment:
Why do you have two separate caches here, but `src/date-range-picker/calendar/grids/index.tsx` uses a single cache for multiple values?

Reply:
The values are passed to dependency arrays so using just one cache might cause a regression. I will check if the code can be further simplified as a follow-up.